### PR TITLE
Fix JQuery vulnerability

### LINKED ISSUE #2406: TestNG 7.3.0 transitive vulnerability CVE-2020-11022 and CVE-2020-11023 due to JQuery 3.4.1
### TestNG Version
latest: 7.3.0 released August 2020

### Explanation
The jquery package in TestNG is vulnerable to the following CVE's:
* CVE-2020-11022
* CVE-2020-11023

The version of JQuery used is 3.4.1, released in May 2019. The fix version is 3.5.0 or later, which was release on April 2020, 3.5.1 in May 2020. It is not clear if there is a specific compatibility reason to not include the latest version of JQuery when TestNG 7.3.0 was released, but maybe the way to include it could be improved. 

#### CVE-2020-11022

CVSSv3: MEDIUM (6.1)

In jQuery versions greater than or equal to 1.2 and before 3.5.0, passing HTML from untrusted sources - even after sanitizing it - to one of jQuery's DOM manipulation methods (i.e. .html(), .append(), and others) may execute untrusted code. This problem is patched in jQuery 3.5.0.

CWE-79 Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')

https://nvd.nist.gov/vuln/detail/CVE-2020-11022

#### CVE-2020-11023

CVSSv3: MEDIUM (6.1)

In jQuery versions greater than or equal to 1.0.3 and before 3.5.0, passing HTML containing <option> elements from untrusted sources - even after sanitizing it - to one of jQuery's DOM manipulation methods (i.e. .html(), .append(), and others) may execute untrusted code. This problem is patched in jQuery 3.5.0.

CWE-79 Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')

https://nvd.nist.gov/vuln/detail/CVE-2020-11023

### Mitigation

In the issue below, similar JQuery vulnerabilities caused the version of TestNG to have that vulnerability as well:
https://github.com/cbeust/testng/issues/2150

It was fixed in this pull request:
https://github.com/cbeust/testng/pull/2153

I can see two mitigations to this issue
* Update dependencies in TestNG to the latest versions before releasing (preferably automated)
* Include JQuery as a dependency, as you include your other project: com.beust:jcommander:jar:1.78:runtime For instance:
https://mvnrepository.com/artifact/org.webjars.bower/jquery

I'm not sure how viable these options are for you, but thank you for your work.

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes #2406

Fixes #2406  .

### Did you remember to?

- [ ] Add test case(s)
- [X] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

File: src/main/java/org/testng/reporters/jq/Main.java [hunk 1]
```diff
@@ -91,18 +87,26 @@ public void generateReport(
...
+        Utils.writeUtf8File(outputDirectory, "index.html", xsb, all);
       }
     } catch (IOException e) {
       Logger.getLogger(Main.class).error(e.getMessage(), e);
     }
   }
+
+  private InputStream load(String fileName) {
+    InputStream is = getClass().getResourceAsStream(Main.TESTNG_RESOURCE_PREFIX + fileName);
+    if (is == null) {
```
Comment (juherr) [inline src/main/java/org/testng/reporters/jq/Main.java:107]: Proposition:

```java
String path;
if (fileName.equals("jquery.min.js") {
  path = "/META-INF/resources/webjars/jquery/3.5.1/jquery.min.js";
} else {
  path = Main.TESTNG_RESOURCE_PREFIX + fileName;
}
return getClass().getResourceAsStream(path);
```

Comment (juherr) [reply src/main/java/org/testng/reporters/jq/Main.java:107]: It will allow us to manage the path strategy and keep the default fallback.

Comment (krmahadevan) [reply src/main/java/org/testng/reporters/jq/Main.java]: fixed

**Comment by krmahadevan** [comment]
@juherr - Please review. Comments have been fixed.

**Comment by krmahadevan** [comment]
ping @juherr - Please let me know if this can be merged

**Review by juherr — APPROVED** [review]

**Pull request merged**